import "alice/matrix/analytics/protos/enum_value_priority_extension.proto";

package NMatrix.NAnalytics.NTest;

message TMatrixAnalyticsValidEnumValueOrdering {
    enum EEnumType {
        VALUE_ZERO_PRIORITY_ZERO = 0 [(NMatrix.NAnalytics.enum_value_priority) = 0];

        VALUE_ONE_PRIORITY_ONE = 1 [(NMatrix.NAnalytics.enum_value_priority) = 1];
        VALUE_TWO_PRIORITY_TWO = 2 [(NMatrix.NAnalytics.enum_value_priority) = 2];

        VALUE_THREE_PRIORITY_FIVE = 3 [(NMatrix.NAnalytics.enum_value_priority) = 5];
        VALUE_FOUR_PRIORITY_FOUR = 4 [(NMatrix.NAnalytics.enum_value_priority) = 4];

        VALUE_FIVE_PRIORITY_THREE = 5 [(NMatrix.NAnalytics.enum_value_priority) = 3];
    }
}

message TMatrixAnalyticsOutOfRangePrioritiesEnumValueOrdering {
    enum EEnumType {
        VALUE_ZERO_PRIORITY_ZERO = 0 [(NMatrix.NAnalytics.enum_value_priority) = 0];

        VALUE_ONE_PRIORITY_ONE = 1 [(NMatrix.NAnalytics.enum_value_priority) = 1];
        VALUE_TWO_PRIORITY_TWO = 2 [(NMatrix.NAnalytics.enum_value_priority) = 2];

        VALUE_THREE_PRIORITY_THREE = 3 [(NMatrix.NAnalytics.enum_value_priority) = 3];
        VALUE_FOUR_PRIORITY_FOUR = 4 [(NMatrix.NAnalytics.enum_value_priority) = 4];

        VALUE_FIVE_PRIORITY_NINE = 5 [(NMatrix.NAnalytics.enum_value_priority) = 9];
    }
}

message TMatrixAnalyticsDuplicatePrioritiesEnumValueOrdering {
    enum EEnumType {
        VALUE_ZERO_PRIORITY_ZERO = 0 [(NMatrix.NAnalytics.enum_value_priority) = 0];

        VALUE_ONE_PRIORITY_ONE = 1 [(NMatrix.NAnalytics.enum_value_priority) = 1];
        VALUE_TWO_PRIORITY_TWO = 2 [(NMatrix.NAnalytics.enum_value_priority) = 2];

        VALUE_THREE_PRIORITY_ONE = 3 [(NMatrix.NAnalytics.enum_value_priority) = 1];
        VALUE_FOUR_PRIORITY_FOUR = 4 [(NMatrix.NAnalytics.enum_value_priority) = 4];

        VALUE_FIVE_PRIORITY_FIVE = 5 [(NMatrix.NAnalytics.enum_value_priority) = 5];
    }
}

message TMatrixAnalyticsMissingPriorityEnumValueOrdering {
    enum EEnumType {
        VALUE_ZERO_PRIORITY_ZERO = 0 [(NMatrix.NAnalytics.enum_value_priority) = 0];

        VALUE_ONE_PRIORITY_ONE = 1 [(NMatrix.NAnalytics.enum_value_priority) = 1];
        VALUE_TWO_PRIORITY_TWO = 2 [(NMatrix.NAnalytics.enum_value_priority) = 2];

        VALUE_THREE_PRIORITY_NONE = 3;
        VALUE_FOUR_PRIORITY_FOUR = 4 [(NMatrix.NAnalytics.enum_value_priority) = 4];

        VALUE_FIVE_PRIORITY_FIVE = 5 [(NMatrix.NAnalytics.enum_value_priority) = 5];
    }
}
